README.md

```diff
@@ -1,38 +1,145 @@
+    
+    // ja bym to rozbił tak (trochę na szybko, więc niektóre rzeczy pominąłem)
+    const workerFilterValueMatch = () => {}
+    
+    const matchWorkersWithFilters = (defaultFilters, selectedFilters, workers) => {
+        return defaultFilters.every((filter) => {
+            const workerFilterValue = worker[filter];
+            const selectedFilterVal = selectedFilters[filter];
 
-Clone this repository and install dependencies inside project directory.
+            if (workerFilterValueMatch(workerFilterValue)) {
+                return selectedFilterVal.some((filterValue) =>
+                    workerFilterValue.includes(filterValue)
+                );
+            }
+            return selectedFilterVal.some(filterValue => filterValue === workerFilterValue);
+        });
+    }
+    
+    const getFilteredWorkers = (defaultFilters, selectedFilters, workers) => {
+        const selected = workers.reduce(
+            (array, worker) => {
+                const hasMatch = matchWorkersWithFilters(defaultFilters, selectedFilters, workers);
+                if (hasMatch) {
+                    return [
+                        ...array,
+                        `${worker.imie} ${worker.nazwisko}`
+                    ]
+                }
+                return array;
+            }, [])
+
+        setFilteredWorkers(selected);
+    };
 
-```
-git clone git@github.com:justFlow5/employee-filter-app.git
-cd employee-filter-app
-npm install
 ```
 
-## Usage
+Pytania:
+- Na przykładzie home albo text-field - poupychane są styled components? 
+- Czasem używane są do definiowania funkcji function a czasem () => , czy jest za tym jakaś logika
 
-To run this project in development mode use:
+- plik TextField.js 
 
+```javascript=
+    const [isActive, setIsActive] = useState('');
+    const [isFilled, setIsFilled] = useState('');
 ```
-npm start
+Czemu tutaj jest string zamiast np. boolean.
+Rozumiem, że niżej jest to używane jako klasa css , ale np clickedOutside jest już boolean i warunkowe wyświetlenie klasy?
+```jsx=
+ <FormField
+            className={`${isFilled} ${isActive} ${
+                isTimeInput ? 'timeType' : ''
+            } ${clickedOutside ? 'inactive' : ''} `}
+        >
 ```
 
----
-
-
-Created with [Create React App](https://github.com/facebook/create-react-app).
-Styled with [Styled-Components](https://github.com/styled-components/styled-components).
+- package.json
+Do czego używany jest jquery? Dlaczego @testing-library jest używane jako dependencies?
+```json
+  "dependencies": {
+    "@testing-library/jest-dom": "^4.2.4",
+    "@testing-library/react": "^9.3.2",
+    "@testing-library/user-event": "^7.1.2",
+    "bootstrap-daterangepicker": "^3.1.0",
+    "jquery": "^3.5.1",
+```
 
-[preview]: https://filter-employee-modal.netlify.app
+- plik helpersFunctions.js
+Else w tych przypadkach nie są potrzebne - ale dlaczego taka funkcja? Głównie chodzi o isTimeInput , dlaczego  tego nie rozbić na dwie mniejsze funkcje?
```

Comment:
Bardzo dobra rada. Zdecydowanie funkcja do rozbicia na dwie.

Reply:
Generalnie jest fajnie, ale bym zdecydowanie podszlifował skille SOLID (to też od Boba Martina). 
Funkcje powinny być małe, najlepiej 10 linii, maksymalnie 20. Jeśli będziesz miał to w głowie to będziesz rozbijał je odruchowo jak tylko zaczną puchnąć. Bardzo dobrą praktyką jest wyrzucanie warunków z if do osobnych funkcji, która zwraca boolean. To się fantastycznie sprawdza w trakcie pisania testów, takie malutkie funkcje, pokryte unit testami robią większość roboty. Jeśli masz dużą funkcję, 40-100 linijkową to pokrycie jej kompletnie testami jest bardzo czasochłonne. Nie mówiąc o tym, że testy też muszą być sensownie napisane. 

Ja u siebie akceptuje maksymalnie 20 linii w jednej funkcji. Średnio pewnie wychodzę poniżej 10. Produkcyjnie jeszcze chyba nigdy nie napisałem dłuższej funkcji :) (pomijając oczywiście render() )